### PR TITLE
Add descriptions for missing errors

### PR DESCRIPTION
**Pull request by ostrokach**

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -497,7 +497,7 @@ rather than a meaningful one. Here are some of the blacklisted names to avoid:
 
 ### Invalid name (C0103) {#C0103}
 
-This error occurs when a name does not follow the [Python Naming Convention] associated with its type (constant, variable, etc.).
+This error occurs when a name does not follow the [Python Naming Convention][PEP8: Naming Conventions] associated with its type (constant, variable, etc.).
```
Comment (david-yz-liu) [inline index.md]: Let's not use the word "type" here, but instead the more generic "role".

File: index.md [hunk 2]
```diff
@@ -1453,14 +1448,415 @@ def check(condition, message):
...
+Corrected version:
+
+```python
+if 'anchovies' in pizza_toppings:  # Error on this line
+    print("Awesome!")
+```
+
+### Literal comparison (R0123) {#R0123}
+
+This error occurs when we use the identity operator `is` to compare Python literals. Whether or not two literals representing the same value (e.g. two identical strings) have the same identity can vary, depending on the way the code is being executed, the code that has ran previously, and the version and implementation of the Python interpreter. For example, each of the following assertions pass if the lines are evaluated together from a Python file, but `assert num is 257` and `assert chars is 'this string fails'` fail if the lines are entered into a Python interpreter one-by-one.
```
Comment (david-yz-liu) [inline index.md]: First sentence: maybe "...to compare non-boolean Python literals"?

File: index.md [hunk 3]
```diff
@@ -1453,14 +1448,415 @@ def check(condition, message):
...
+~~~~
+
+To prevent the confusion, it is advisable to use the equality operator `==` when comparing objects with Python literals.
+
+```python
+num = 256
+assert num == 256
+
+num = 257
+assert num == 257  # Assertion fails if typed into a Python interpreter
```
Comment (david-yz-liu) [inline index.md]: This comment and the one below are weird. Doesn't `num is 257` fail, while `num == 257` succeed?

File: index.md [hunk 4]
```diff
@@ -1453,14 +1448,415 @@ def check(condition, message):
...
+
+**See also**:
+
+- [Literally Literals and Other Number Oddities In Python]
+- [StackOverflow: About the changing id of an immutable string]
+- [StackOverflow: When does Python allocate new memory for identical strings?]
+
+### Unsupported assignment operation (E1137) {#E1137}
+
+This error occurs when we assign something to an object which does not support assignment (i.e. an object which does not define the `__setitem__` method).
```
Comment (david-yz-liu) [inline index.md:1520]: I realized I don't like the code example or description here. I want you to be explicit about *attribute* assignment as the issue -- slicing introduces an extra level of complexity that's unnecessary. I imagine the following code example could work better:

```python
x = 3
x.new_attr = 'hello'  # Error on this line
```

You don't need to include a description of how to "fix" this problem.

File: index.md [hunk 5]
```diff
@@ -1453,14 +1448,415 @@ def check(condition, message):
...
+
+```python
+my_string = "Hello World!"
+my_string = my_string[:6] + "Universe!"
+print(my_string)  # Prints 'Hello Universe!'
+```
+
+### Expression not assigned (W0106) {#W0106}
+
+This error occurs when an expression that is not a function call is not assigned to a variable. Typically, this indicates that we were indenting to do something else.
```
Comment (david-yz-liu) [inline index.md]: Last part should be "intending" not "indenting"

File: index.md [hunk 6]
```diff
@@ -1453,14 +1448,415 @@ def check(condition, message):
...
+
+Corrected version:
+
+```python
+print_greeting(name="Arthur")
+```
+
+### Redefined argument from local (R1704) {#R1704}
+
+This error occurs when a local name is redefining the name of a parameter. We should avoid reusing the name of a parameter in binding operations such as for iteration, with statement assignment, and exception handler assignment, as this needlessly obfuscates the code.
```
Comment (david-yz-liu) [inline index.md]: I think you can remove the second sentence (the terminology isn't helpful here), and also reference the other name shadowing-type checks.

File: index.md [hunk 7]
```diff
@@ -1453,14 +1448,415 @@ def check(condition, message):
...
+~~~~
+
+Corrected version:
+
+```python
+def pos(temp: int) -> str:
+    if temp > 0:
+      return 'positive'
+    else:
+        return 'negative'
```
Comment (david-yz-liu) [inline index.md]: Indentation is off here.

**Review by david-yz-liu — CHANGES_REQUESTED** [review]
@ostrokach since you're still doing some work, I've made some comments here.

File: index.md (hunk 3 above)
Comment (ostrokach) [reply index.md]: Yes, sorry. I copy-pasted and forgot to remove the comments.

File: index.md (hunk 4 above)
Comment (ostrokach) [reply index.md:1520]: As far as I understand, this error only works for item assignment? I looked at the [examples pylint uses to test their code](https://github.com/PyCQA/pylint/blob/master/pylint/test/functional/unsupported_assignment_operation.py), and all of them are item assignment. I don't think the example you give throws this error.

I did remove the code for how to "fix" the problem.

Comment (david-yz-liu) [reply index.md:1520]: No you're right, my bad. I was confusing `__setitem__` with `__setattr__`.

**Review by david-yz-liu — APPROVED** [review]

**Pull request closed without merging**